Front end tests fail in Phantom

```
» npm run-script test

> fxa-content-server@0.0.1 test /Users/stomlinson/development/fxa-content-server
> intern-runner config=tests/intern

Defaulting to "runner" reporter
Listening on 0.0.0.0:9090
Initialised phantomjs 1.9.7 on mac-unknown-64bit
Waiting for email...
Waiting for email...
Waiting for email...
passes: 151failures: 2duration: 13.74s
lib/channel/web
send
is a standin that does nothing
‣
lib/translator
get
returns translated string when it exists
‣
returns untranslated string when translation does not exist
‣
can do string interpolation on unnamed `%s` when given array context
‣
can do string interpolation on named `%(name)s` when given array context
‣
can do interpolation multiple times with an array
‣
can do interpolation multiple times with an object
‣
does no replacement on %s and %(name)s if not in context
‣
leaves remaining %s if not enough items in context
‣
lib/xss
href
allows http href
‣
allows https href
‣
allows href with query parameters
‣
allows but escapes URLs that try to break out
‣
disallows javascript: href
‣
disallows href without a scheme
‣
disallows relative scheme
‣
disallows data URI scheme
‣
only allows strings
‣
allows hrefs of the max length
‣
disallowed hrefs that are too long
‣
lib/session
set
can take a key value pair
‣
can take an object
‣
will not overwrite items in Session.prototype
‣
clear
with a key clears item
‣
with no key clears all items
‣
will not clear items in Session.prototype
‣
will not clear items in DO_NOT_CLEAR
‣
load
loads data from localStorage
‣
does not load up items in DO_NOT_PERSIST
‣
lib/url
searchParam
returns a parameter from window.location.serach, if it exists
‣
returns undefined if parameter does not exist
‣
does not throw if str override is not specified
‣
lib/channel/fx-desktop
init
sends the user to the settings page if signed in
‣
sends the user to the signup page if not signed in
‣
send
sends a message to the browser
‣
retries sending until a response is received from the browser51ms
‣
times out if browser does not respond114ms
‣
does not except on timeout if callback is not given500ms
‣
on
registers a callback to be called when the browser sends the registered message
‣
views/sign_in
constructor creates it
is drawn
‣
updatePasswordVisibility
pw field set to text when clicked
‣
pw field set to password when clicked again
‣
isValid
returns true if both email and password are valid
‣
returns false if email is invalid
‣
returns false if password is invalid
‣
showValidationErrors
shows an error if the email is invalid
‣
shows an error if the password is invalid
‣
views/sign_in used for /force_auth without email
prints an error message
‣
views/sign_in used for /force_auth?email="testuser@testuser.com"
does not print an error message
‣
does not allow the email to be edited
‣
user cannot create an account
‣
isValid is successful when the password is filled out
‣
views/settings
with no session
redirects to signin
‣
with session
signOut
"before each" hook
‣
Error: timeout of 5000ms exceeded
views/sign_up
isValid
returns true if email, password, and age are all valid
‣
returns false if email is empty
‣
returns false if email is not an email address
‣
returns false if password is empty
‣
returns false if password is invalid
‣
returns false if age is invalid
‣
showValidationErrors
shows an error if the email is invalid
‣
shows an error if the password is invalid
‣
shows an error if no year is selected
‣
submit
sends the user to confirm screen if form filled out, >= 14 years ago
‣
Error: timeout of 5000ms exceeded
sends the user to cannot_create_account screen if user selects <= 13 years ago
‣
sends user to cannot_create_account when visiting sign up if they have already been sent there
‣
updatePasswordVisibility
pw field set to text when clicked
‣
pw field set to password when clicked again
‣
lib/fxa-client
signUp/signUpResend
signUp signs up a user with email/password88ms
‣
informs browser of customizeSync option68ms
‣
signUpResend resends the validation email99ms
‣
signIn
signin with unknown user should call errorback
‣
signs a user in with email/password115ms
‣
informs browser of customizeSync option106ms
‣
passwordReset/passwordResetResend
requests a password reset80ms
‣
signOut
signs the user out76ms
‣
changePassword
changes the user's password198ms
‣
deleteAccount
deletes the user's account143ms
‣
views/base
render
renders the template without attaching it to the body
‣
afterVisible
focuses descendent element containing `autofocus` if html has `no-touch` class
‣
does not focus descendent element containing `autofocus` if html does not have `no-touch` class
‣
displayError
translates and display an error in the .error element
‣
hides any previously displayed success messages
‣
displaySuccess
translates and display an success in the .success element
‣
hides any previously displayed error messages
‣
navigate
navigates to a page
‣
focus
focuses an element
‣
BaseView.preventDefaultThen
can take the name of a function as the name of the event handler
‣
can take a function as the event handler
‣
can take no arguments at all
‣
views/form
enableSubmitIfValid
enables submit button if isValid returns true
‣
disabled submit button if isValid returns false
‣
validateAndSubmit
submits form if isValid returns true
‣
shows validation errors if isValid returns false
‣
showValidationError
sets the title of the nearest tooltip
‣
removes the title of the nearest tooltip when changed
‣
focuses the invalid element
‣
adds invalid class to the invalid element
‣
invalid class is removed as soon as element is valid again
‣
views/complete_sign_up
constructor creates it
is drawn
‣
afterRender
shows an error if uid is not available on the URL
‣
throws an error if code is not available on the URL
‣
attempts to complete verification if code and uid are available
‣
shows redirectTo button if redirectTo is available
‣
does not show redirectTo button if redirectTo is not available
‣
views/confirm
constructor creates it
is drawn
‣
resend
resends the confirmation email107ms
‣
resend only occurs when clicking on #resend79ms
‣
views/tos
Back button displayed if Session.canGoBack is true
‣
Back button not displayed if Session.canGoBack is false
‣
views/change_password
with no session
redirects to signin
‣
with session
isValid
returns true if both old and new passwords are valid and different
‣
returns true if both old and new passwords are valid and the same
‣
returns false if old password is too short
‣
returns false if new password is too short
‣
showValidationErrors
shows an error if the password is invalid
‣
shows an error if the new_password is invalid
‣
submit
prints an error message if both passwords are the same
‣
changes from old to new password, redirects user to signin125ms
‣
views/delete_account
with no session
redirects to signin
‣
with session
isValid
returns true if email and password are filled out
‣
returns false if not an email
‣
returns false if password is too short
‣
showValidationErrors
shows an error if the email is invalid
‣
shows an error if the password is invalid
‣
submit
deletes the users account, redirect to signup
‣
views/confirm_reset_password
constructor creates it
is drawn
‣
resend
resends the confirmation email93ms
‣
resend only occurs when clicking on #resend77ms
‣
views/complete_reset_password
constructor creates it
is drawn
‣
updatePasswordVisibility
pw field set to text when clicked
‣
pw field set to password when clicked again
‣
isValid
returns true if token, code, email, password, vpassword available, password & vpassword valid and the same
‣
returns false if token, code, email, password, vpassword available, password & vpassword valid are different
‣
returns false if token, code, email, password, vpassword available, password invalid
‣
returns false if token, code, email, password, vpassword available, vpassword invalid
‣
returns false if email missing
‣
returns false if code missing
‣
returns false if token missing
‣
showValidationErrors
shows an error if the password is invalid
‣
shows an error if the vpassword is invalid
‣
views/pp
Back button displayed if Session.canGoBack is true
‣
Back button not displayed if Session.canGoBack is false
‣
views/reset_password
constructor creates it
is drawn
‣
isValid
returns true if email address is entered
‣
returns false if email address is empty
‣
returns false if email address is invalid
‣
showValidationErrors
shows an error if the email is invalid
‣
views/reset_password_complete
render
renders
‣
shows redirectTo link and service name if available
‣
does not show redirectTo link if unavailable
‣
lib/router
navigate
Tells the router to havigate to a page
‣
preserves window search parameters across screen transition
‣
2
Test main - mocha tests - run the mocha tests FAILED on phantomjs 1.9.7 on mac-unknown-64bit:
Error: expected '2' to equal '0'
AssertionError: expected '2' to equal '0'
    at /Users/stomlinson/development/fxa-content-server/tests/functional/mocha.js:43:20
    at /Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/lib/wd.js:924:33
    at signalListener (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:37:21)
    at signalWaiting (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:28:4)
    at resolve (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:192:5)
    at signalDeferred (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:81:15)
    at signalListener (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:52:6)
    at signalWaiting (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:28:4)
    at resolve (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:192:5)
    at signalDeferred (/Users/stomlinson/development/fxa-content-server/node_modules/intern-geezer/node_modules/dojo/Deferred.js:81:15)
No coverage report available
phantomjs 1.9.7 on mac-unknown-64bit: 1/24 tests failed

----------+-----------+-----------+-----------+-----------+
File      |   % Stmts |% Branches |   % Funcs |   % Lines |
----------+-----------+-----------+-----------+-----------+
----------+-----------+-----------+-----------+-----------+
All files |       100 |       100 |       100 |       100 |
----------+-----------+-----------+-----------+-----------+

TOTAL: tested 1 platforms, 1/24 tests failed
npm ERR! weird error 1
npm ERR! not ok code 0
```

--- a/app/tests/spec/lib/session.js
+++ b/app/tests/spec/lib/session.js
@@ -21,6 +21,10 @@ function (mocha, chai, Session) {
 
     afterEach(function () {
       Session.clear();
+      // Since Session is a singleton and channel is not cleared, attaching
+      // the mock Session.channel can interfere with other tests, depending
+      // on the ordering of the tests.
+      delete Session.channel;
     });
 
     describe('set', function () {

